iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/helplets.inl

```diff
@@ -113,7 +119,7 @@ inline bool isValidPathToDirectory(const string<StringCapacity>& name) noexcept
         // //some///fuu////bar
         if (separatorPosition && *separatorPosition == 0)
         {
-            temp = *temp.substr(*separatorPosition + 1);
+            temp.substr(*separatorPosition + 1).and_then([&](auto& s) { temp = s; });
             continue;
         }
```

Comment:
I struggled a bit with this and thought there would be an infinite loop for `"foo/"`. Just as food for thought. Would it make sense to move this into `if (separatorPosition)` below and check within that if statement with `if (*separatorPosition == 0) {...} else {...}`?

...

After some thinking, isn't this an infinite loop ... wait, `substr` returns a `nullopt` only if `pos` is > `size`. So `substr(1)` from `string<1> foo {"f"};` returns an empty string?

Not sure if there should be an abort in the `or_else` branch? It should never be reached, though. This would make MC/DC testing harder. Maybe we can have an `unreachable` macro?

Reply:
Yes, in your example `substr` would return an empty string. AFAIU this happens either in line 139 (`"foo/"`) or in line 122 (`"f"`) and then leads to leaving the loop. So I think there is no else branch needed.